### PR TITLE
Avoid create index label on property which contains primary key

### PR DESCRIPTION
**Pull request by Linary**
Improve #238

Change-Id: I902b18c33260f142eb099223509c2d09dc781523

### CODE REVIEW

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java [hunk 1]
```diff
@@ -530,6 +530,37 @@ public void testAddIndexLabelSecondaryPrefixWithExistedRange() {
...
     }
 
+    @Test
+    public void testAddIndexLabelOnPrimaryKeyProps() {
+        super.initPropertyKeys();
+        SchemaManager schema = graph().schema();
+
+        schema.vertexLabel("person").properties("name", "age", "city")
+              .primaryKeys("name").create();
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java]: add a empty line before assert

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java [hunk 2]
```diff
@@ -530,6 +530,37 @@ public void testAddIndexLabelSecondaryPrefixWithExistedRange() {
...
+                  .by("name").create();
+        });
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            schema.indexLabel("studentByNameAndAge").onV("student").secondary()
+                  .by("name", "age").create();
+        });
+
+        schema.vertexLabel("student").properties("name", "age", "city")
+              .primaryKeys("name", "age").create();
+        schema.indexLabel("studentByName").onV("student").secondary()
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java]: ditto

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java [hunk 3]
```diff
@@ -530,6 +530,37 @@ public void testAddIndexLabelSecondaryPrefixWithExistedRange() {
...
+              .by("name").create();
+
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            schema.indexLabel("studentByNameAndAge").onV("student").secondary()
+                  .by("name", "age").create();
+        });
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            schema.indexLabel("studentByAgeAndName").onV("student").secondary()
+                  .by("age", "name").create();
+        });
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java]: add case: `by("name", "age", "city")`

File: hugegraph-core/src/main/java/com/baidu/hugegraph/schema/builder/IndexLabelBuilder.java [hunk 4]
```diff
@@ -345,6 +346,19 @@ private void checkFields(Set<Id> propertyIds) {
     }
 
     private void checkRepeatIndex(SchemaLabel schemaLabel) {
+        if (schemaLabel instanceof VertexLabel) {
+            VertexLabel vl = (VertexLabel) schemaLabel;
+            if (vl.idStrategy().isPrimaryKey()) {
+                List<String> newFields = this.indexFields;
+                List<String> pkFields = this.transaction.graph()
+                                            .mapPkId2Name(vl.primaryKeys());
+                E.checkArgument(!newFields.containsAll(pkFields),
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/schema/builder/IndexLabelBuilder.java]: this.indexFields

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java [hunk 5]
```diff
@@ -530,6 +530,43 @@ public void testAddIndexLabelSecondaryPrefixWithExistedRange() {
...
+
+        schema.vertexLabel("person").properties("name", "age", "city")
+              .primaryKeys("name").create();
+
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            schema.indexLabel("personByName").onV("person").secondary()
+                  .by("name").create();
+        });
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            schema.indexLabel("studentByNameAndAge").onV("student").secondary()
```
Comment (zhoney) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java]: personByNameAndAge

**Pull request merged**